### ISSUE TITLE
Save sparse weights in mesh tfidf-svm

### ISSUE THREAD

**Issue by nsorros** [wellcometrust/grants_tagger#114]
`SGDClassifier` https://scikit-learn.org/stable/modules/generated/sklearn.linear_model.SGDClassifier.html which is what we use under the hood for `MeshTfidfSVM` provides a sparsity function that creates sparse weights which results in a much smaller model size. To effectively use that, it would be better to zero weights below a value say `0.001` that do not contribute much.

As a reminder the disease mesh model ended up being 70GB and if we trained on the entire mesh it would hundreds of GB. Weight pruning is a commonly used strategy to reduce model size.

Note that to use the weights for predictions you need to make them `dense` through `densify` so some care is still needed to fit into memory. Batching predictions by label as we do now will still be a good way to deal with that.
